gui/kitchen-info.lua

```diff
@@ -0,0 +1,414 @@
+-- Adds more info to the Kitchen screen.
+--@ enable = true
+--[====[
+
+gui/kitchen-info
+===================
+Adds more info to the Kitchen screen.
+
+Usage::
+
+    gui/kitchen-info enable|disable|help
+    enable|disable gui/kitchen-info
+
+]====]
+
+gui = require 'gui'
+dialogs = require 'gui.dialogs'
+gps = df.global.gps
+world = df.global.world
+
+local kitchen_overlay = defclass(kitchen_overlay, gui.Screen)
+kitchen_overlay.focus_path = 'kitchen_overlay'
+
+local show_processing = true
+
+local COLUMNS = {
+    {
+        label = "Extract",
+        maxValueLength = 6,
+        minValueLeftPadding = 0,
+        emptyValueLeftPadding = 1,
+        activeOnPages = { 'VegetablesFruitLeaves' }
+    },
+    --[[
+    -- Hide this column because the only plant with BAG_ITEM reaction is Quarry Bush, which is neither cookable nor brewable so it never appears in the Kitchen screen
+    {
+        label = "ToBag",
+        maxValueLength = 5,  -- actually 6 but rarely appears
+        minValueLeftPadding = 0,
+        emptyValueLeftPadding = 0,
+        activeOnPages = { 'VegetablesFruitLeaves' }
+    },
+    --]]
+    {
+        label = "Thread",
+        maxValueLength = 6,
+        minValueLeftPadding = 0,
+        emptyValueLeftPadding = 1,
+        activeOnPages = { 'VegetablesFruitLeaves' }
+    },
+    {
+        label = "Mill",
+        maxValueLength = 5,
+        minValueLeftPadding = 0,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    },
+    --[[
+    -- Unimplemented
+    {
+        label = "Press",
+        maxValueLength = 5,
+        minValueLeftPadding = 0,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    },
+    --]]
+    {
+        label = "EatRaw",
+        maxValueLength = 4,
+        minValueLeftPadding = 1,
+        emptyValueLeftPadding = 1,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    }
+}
+
+local COLUMNS_BY_LABEL = {}
+
+local function ArrayToSet (array)
+    local set = {}
+    for _, value in ipairs (array) do
+        set[value] = true
+    end
+    return set
+end
```

Comment:
this can be replaced with a call to `utils.invert()`

Reply:
Done.